### DISCUSSION TITLE
Code documentation for quicker insights and understandings

### DISCUSSION THREAD

**Discussion by klutzzykitty** [blakeblackshear/frigate#4470] [General]
The current `docs` are excellent to understand the procedures involved in setting up Frigate as well as some general concepts on it's 
internal workings. 

_I'm curious as to whether documentations on the code part and it's working exist._ 

**If so, can it be shared or is it already available ?** 
As someone thinking of contributing towards the AI and computer vision part of Frigate, It would be useful for _devs_  to 
understand code quickly and know what part does what, rather than spending time on figuring it out by themselves. 

Thanks ! : )

**Comment by NickM-27** [comment]
What's in the docs is what is there. However, I started contributing a little less than a year ago and I found that the code internally is very clean and straightforward so it was a lot easier than other projects to contribute to. If you have any questions I'd be happy to answer them. Dev docs could be a source of future improvements.

**Reply by klutzzykitty** [reply]
Thanks @NickM-27 
The code is indeed clean and up-to mark. 

Currently, one has to spend some time (varies on how good they are in it) getting to know the Intricacies and also understand why it was programmed that way before thinking of improving it in any way, lets say, by optimising some functionality through better implementation techniques. 

Presence of some notes or dev docs or even a basic overall explanation of code execution flow would lessen the time and also help the person to not concentrate on understanding other parts that might not have been of interest, isn't it ?  

You mentioned you started contributing to it a year ago. 
You'd have made some notes or two on some aspects of its programming. Even that can suffice if I may request you to share it. 

I can make a dev doc of Frigate's working later on as it doesn't exist yet.

**Reply by NickM-27** [reply]
To be honest I haven't made notes, I contribute enough that it has stayed fresh. 

I do agree a wireframe could be helpful to do this, might be worthwhile after 0.12 since there's a decent bit of structure changes going on.

**Reply by klutzzykitty** [reply]
Sounds good. 

Any idea on when V0.12. is planned to be released or when its corresponding beta releases are planned ?

**Reply by NickM-27** [reply]
You can watch the progress here: https://github.com/blakeblackshear/frigate/pull/4055

no idea on the timeline

**Reply by klutzzykitty** [reply]
Oh that's cool. 
Was wondering where the discussion for release 0.12 was. 
Thanks for pointing it out. That'll help.
